web/modules/custom/rdf_entity/src/Entity/Query/Sparql/SparqlCondition.php

```diff
@@ -4,47 +4,785 @@
+        $condition['field']->compile($query);
+        continue;
+      }
+      // The id and bundle are set in the constructor.
+      else {
+        $mappings = $this->fieldHandler->getFieldPredicates($entity_type->id(), $condition['field'], $condition['column']);
+      }
+
+      $field_name = $condition['field'] . '__' . $condition['column'];
+      if (count($mappings) === 1) {
+        $this->fieldMappings[$field_name] = reset($mappings);
+      }
+      else {
+        if (!isset($this->fieldMappings[$field_name])) {
+          $this->fieldMappings[$field_name] = $field_name . '_predicate';
+        }
+        // The predicate mapping is not added as a direct filter. It is being
+        // loaded by the database. There is no way that in a single request, the
+        // same predicate is found with a single and multiple mappings.
+        // There is no filter per bundle in the query.
+        $this->fieldMappingConditions[] = [
+          'field' => $field_name . '_predicate',
+          'value' => array_values($mappings),
+          'operator' => 'IN',
+        ];
+      }
+    }
+  }
+
+  /**
+   * Returns the string version of the conditions.
+   *
+   * @return string
+   *   The string version of the conditions.
+   */
+  public function toString() {
+    // In case of re-compiling, remove previous fragments. This will ensure that
+    // not previous duplicates or leftovers remain.
+    $this->conditionFragments = [];
+
+    // There are cases where not a single triple is formed. In these cases, a
+    // default condition has to be added in order to retrieve distinct Ids.
+    if ($this->needsDefault) {
+      $this->addDefaultCondition();
+    }
+
+    // The fieldMappingConditions are added first because they are converted
+    // into a 'VALUES' clause which increases performance.
+    $this->fieldMappingConditionsToString();
+
+    // Convert the conditions.
+    $this->conditionsToString();
+
+    // Put together everything.
+    $condition_fragments = array_merge($this->tripleFragments, $this->conditionFragments);
+    $this->stringVersion = implode(self::$conjunctionMap[$this->conjunction]['delimeter'], array_unique($condition_fragments));
+    return $this->stringVersion;
+  }
+
+  /**
+   * Converts the field mapping conditions into string versions.
+   */
+  private function fieldMappingConditionsToString() {
+    foreach ($this->fieldMappingConditions as $condition) {
+      $field_name = $condition['field'] . '__' . $condition['column'];
+      $field_predicate = $this->fieldMappings[$field_name];
+      $this->addConditionFragment(self::ID_KEY . ' ' . $this->escapePredicate($field_predicate) . ' ' . $this->toVar($field_name));
+      $condition['value'] = SparqlArg::toResourceUris($condition['value']);
+      $this->addConditionFragment($this->compileValuesFilter($condition));
+    }
+  }
+
+  /**
+   * Converts the conditions into string versions.
+   */
+  private function conditionsToString() {
```

Comment:
Well I was thinking something like a procedure to extract the pieces of the pattern separately but that takes a bit of time to design and implement and I could not spend time. If you have a suggestion we can discuss on that.